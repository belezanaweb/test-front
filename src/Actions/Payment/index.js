export const saveProduct = (data) => {
  return {
    type: '@reducer/ADD_PRODUCTS',
    payload: data
  }
}

export const saveTotal = (data) => {
  return {
    type: '@reducer/ADD_TOTALS',
    payload: data
  }
}
