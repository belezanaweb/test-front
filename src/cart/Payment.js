import React, { Component } from 'react';
import { connect } from 'react-redux';
import * as actionCreators from '../store/actions/index.js';
import FinalizeAccount from '../components/FinalizeAccount';
import Header from '../components/Header';
import Button from '../components/Button';
import UserForm from '../components/UserForm'

class Payment extends Component {
  render() {
    let isDisabled =
      !this.props.cardName ||
      !this.props.cardCvv ||
      !this.props.cardDate ||
      !this.props.cardNumber;

    return (
      <div>
        <Header />
        <UserForm 
          cardName={this.props.cardName} 
          cardCvv={this.props.cardCvv} 
          cardDate={this.props.cardDate} 
          cardNumber={this.props.cardNumber} 
        />
        <FinalizeAccount
          subTotal={this.props.subTotal}
          shippingTotal={this.props.shippingTotal}
          discount={this.props.discount}
          total={this.props.total}
        />
        <Button
          label='FINALIZAR O PEDIDO'
          handleClick='success'
          disabled={isDisabled}
        />
      </div>
    );
  }

}
const mapStateToProps = (state) => {
  return state
};

export default connect(mapStateToProps, actionCreators)(Payment);
