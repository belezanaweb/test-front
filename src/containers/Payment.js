import React, { Component } from 'react';
import { connect } from 'react-redux';
import { Field, reduxForm } from 'redux-form';
import styled from 'styled-components';

const Input = styled.input`
    border: 1px solid #ddd;
    padding-left: 5px;
    width: 300px;
    transition: 0.2s cubic-bezier(0.68, -0.55, 0.265, 1.55) all;

    &.error {
        border-left: 8px solid #D0021B;
    }

    &.pass {
        border-left: 8px solid #7ED321;
    }

    &:focus {
        border-left: 8px solid #449CFA;
    }
`

class Payment extends Component {
    constructor(props) {
        super(props);
        this.state = {}
    }

    renderField = (field) => {
        console.log('what is field', field)
        const { meta: { touched, error }, input: { value }, label } = field;

        const inputState = () => {
            if (error) {
                return 'error'
            } else if (touched && value) {
                return 'pass'
            } else {
                return ''
            }
        }
        
        return (
            <div>
                <label>{label}:</label>
                <Input className={inputState()} placeholder={field.placeholder} {...field.input} />
            </div>
        )
    }

    onSubmit = (values) => {
        console.log('values', values)
        this.props.history.push('/confirmation')
    }

    render() {
        const { handleSubmit } = this.props;
        return (
            <form onSubmit={handleSubmit(this.onSubmit)}>
                <Field name="cardNumber" label="Número do cartão" placeholder="____ ____ ____ ____" component={this.renderField} />
                <Field name="personName" label="Nome do titular" placeholder="Como no cartão" component={this.renderField} />
                <Field name="expire" label="Validade" placeholder="__ / ____" component={this.renderField} />
                <Field name="cvv" label="CVV" placeholder="___" component={this.renderField} />
                <button type="submit">Finalizar Pedido</button>
            </form>
        )
    }
}

const validate = (values) => {
    const errors = {}
    if (!values.cardNumber) {
        errors.cardNumber = "Enter a card number!"
    }
    if (!values.personName) {
        errors.personName = "Enter a person name!"
    }
    if (!values.expire) {
        errors.expire = "Enter a expire date!"
    }
    if (!values.cvv) {
        errors.cvv = "Enter a cvv!"
    }

    return errors
} 

const mapStateToProps = (state) => ({
    data: state.fetchedData.items,
    loading: state.fetchedData.loading,
    error: state.fetchedData.error
})

Payment = connect(mapStateToProps)(Payment);

export default reduxForm({
    validate,
    form: 'PaymentForm',
    destroyOnUnmount: false
})(Payment);