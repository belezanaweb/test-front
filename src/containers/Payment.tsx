import * as React from 'react';
import { Connection, Inject } from 'exredux';
import { modelStore } from '../service/modelStore';
import { ProductListModel } from '../service/models/ProductListModel';
import { Title } from '../components/Title';
import { CartSummary } from '../components/CartSummary';
import { Button } from '../components/Button';
import { CreditCardForm } from './CreditCardForm';
import { CardModel } from '../service/models/CardModel';
import { Wrapper } from '../components/Wrapper';
import { Link } from 'react-router-dom';
// ---------------------------------------------------------------------
// --- COMPONENT -------------------------------------------------------
// ---------------------------------------------------------------------
class Props {
  @Inject productList?: ProductListModel;
  @Inject cardModel?: CardModel;
}

@Connection({
  modelStore,
  props: Props
})
export class Payment extends React.Component<Props> {
  render() {
    const { productList, cardModel } = this.props;
    const products = productList.response.data;
    return (
      <Wrapper>
        <Title>CARTÃO DE CRÉDITO</Title>
        <CreditCardForm />
        <CartSummary
          discount={products.discount}
          shippingTotal={products.shippingTotal}
          subTotal={products.subTotal}
          total={products.total}
        />

        <Link to="success">
          <Button disabled={!cardModel.isValidAll()}>Finalizar o Pedido</Button>
        </Link>
      </Wrapper>
    );
  }
}
