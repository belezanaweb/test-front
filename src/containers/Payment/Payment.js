import React, { useContext, useState, useRef } from 'react'
import { useHistory } from 'react-router-dom'
import { PaymentContext } from '../../store/PaymentStore'
import { Button } from '../../components/Button/Button'
import { PaymentForm } from '../../components/PaymentForm/PaymentForm'
import { PurchaseSummary } from '../../components/PurchaseSummary/PurchaseSummary'
import { Stepper } from '../../components/Stepper/Stepper'
import { Layout, LayoutGrid, LayoutAside } from '../../components/Layout/Layout'

const Payment = () => {
  const formRef = useRef()
  const history = useHistory()
  const paymentContext = useContext(PaymentContext)
  const [isValid, setIsValid] = useState(false)

  const onChangeHandler = () => {
    setIsValid([...formRef.current.elements].every((el) => el.checkValidity()))
  }

  const onSubmitHandler = (event) => {
    event.preventDefault()

    if (!isValid) return

    const formData = new FormData(formRef.current)

    const ccNumber = formData.get('cardnumber')
    const ccName = formData.get('ccname')
    const ccExpiry = formData.get('cc-exp')

    paymentContext.dispatch.ccNumber(ccNumber)
    paymentContext.dispatch.ccName(ccName)
    paymentContext.dispatch.ccExpiry(ccExpiry)

    history.push('/confirmacao')
  }

  return (
    <Layout>
      <Stepper steps={steps} />
      <LayoutGrid>
        <PaymentForm ref={formRef} onSubmit={onSubmitHandler} onChange={onChangeHandler} />
        <LayoutAside>
          <PurchaseSummary {...summary} />
          <Button type="submit" form="_form" disabled={!isValid}>
            Finalizar o pedido
          </Button>
        </LayoutAside>
      </LayoutGrid>
    </Layout>
  )
}

const steps = [
  {
    title: 'Sacola'
  },
  {
    active: true,
    title: 'Pagamento'
  },
  {
    title: 'Confirmação'
  }
]

const summary = {
  subTotal: 624.8,
  shippingTotal: 5.3,
  discount: 30,
  total: 618.9
}

export default Payment
