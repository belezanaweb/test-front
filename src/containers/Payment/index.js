import React from "react";
import MenuTabs from "../../components/MenuTabs"

const Payment = () => {
    return (
        <div>
            <MenuTabs />
            <h4>Payment</h4>
        </div>
    )
}

export default Payment;