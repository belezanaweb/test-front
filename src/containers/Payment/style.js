import styled from 'styled-components';

export const Container = styled.div`
    display: flex;
    flex-direction: column;
    align-items: center;
    height: 100%;

    @media (min-width: 1024px) {
        display: flex;
        justify-content: center;
        flex-direction: row;
        align-items: flex-start;
  }
`;

export const InputBigger = styled.input`
    box-sizing: border-box;
    height: 45px;
    width: 320px;
    border-radius: 3px;
    background-color: #FFF;
    border: 1px solid #E7E7E7;
    padding-left: 1vh;
    box-shadow: inset 0 1px 2px 0 rgba(0,0,0,0.2);
    font-family: "Helvetica Neue";
    font-size: 16px;
    line-height: 19px;
    ::placeholder {
        color: #E0E7EE;
    }
`;

export const BoxBigger = styled.div`
    padding: 1vh;
`;

export const BoxSmaller = styled.div`
    display: flex;
    padding: 1vh;
`;

export const LabelBigger = styled.label`
    height: 14px;
    width: 129.71px;
    color: #CCC;
    font-family: "Helvetica Neue";
    font-size: 12px;
    font-weight: 700;
    line-height: 14px;
`;

export const LabelSmaller = styled.label`
    height: 64px;
    width: 160px;
    color: #CCC;
    font-family: "Helvetica Neue";
    font-size: 12px;
    font-weight: 700;
    line-height: 14px;
`;

export const InputSmaller = styled.input`
    box-sizing: border-box;
    height: 45px;
    width: 140px;
    border-radius: 3px;
    background-color: #FFF;
    border: 1px solid #E7E7E7;
    padding-left: 1vh;
    box-shadow: inset 0 1px 2px 0 rgba(0,0,0,0.2);
    ::placeholder {
            color: #E0E7EE;
    }
`;
