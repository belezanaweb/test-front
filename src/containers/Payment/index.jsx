/* eslint-env browser */
import React, { useEffect, useState } from 'react';
import { connect, useDispatch, useSelector } from 'react-redux';
import { Link, Redirect } from 'react-router-dom';

import { Container, Row, Col } from 'react-grid-system';

import DefaultLayout from 'layouts/default';

import Button from 'components/Button';
import Text from 'components/Text';
import Checkout from 'components/Checkout';

import Card from 'components/Card';
import Input from 'components/Input';

import { setCardData } from 'store/cart/actions';
import ContainerStyled from '../style';

const Payment = () => {
  useEffect(() => {
    document.title = `Beleza na Web - Pagamento`;
  }, []);
  const dispatch = useDispatch();
  const data = useSelector(state => state.cart.data);

  const [cardNumber, setCardNumber] = useState();
  const [cardName, setCardName] = useState(undefined);
  const [cardValidity, setCardValidity] = useState(undefined);
  const [cardError, setCardError] = useState(undefined);
  const [ccv, setCcv] = useState(undefined);

  const handleFormValid = () => {
    if (
      !cardNumber ||
      cardNumber.length < 19 ||
      !ccv ||
      ccv.length < 3 ||
      !cardValidity ||
      cardValidity.length < 7 ||
      !cardName
    ) {
      return true;
    }
    return false;
  };

  useEffect(() => {
    const mc = /^5[1-5][0-9]{14}$|^5[1-5][0-9.X]{17}$/;
    const vi = /^4[0-9]{12}(?:[0-9]{3})?$|^4[0-9.X]{15}(?:[0-9]{3})?$/;
    const dc = /^3(?:0[0-5]|[68][0-9])[0-9]{11}$|^3(?:0[0-5]|[68][0-9])[0-9.X]{13}$/;

    if (typeof cardNumber !== 'undefined' && mc.test(cardNumber)) {
      return setCardError(true);
    }
    if (typeof cardNumber !== 'undefined' && vi.test(cardNumber)) {
      return setCardError(true);
    }
    if (typeof cardNumber !== 'undefined' && dc.test(cardNumber)) {
      return setCardError(true);
    }
    return setCardError(false);
  }, [cardNumber, setCardError]);

  const handleSubmit = () => {
    const card = {
      cardNumber,
      cardName,
      cardValidity,
      ccv,
    };
    dispatch(setCardData(card));
  };

  return (
    <ContainerStyled className="animated fadeIn faster">
      <Container>
        {data && data.items ? (
          <Row>
            <Col lg={8} xs={12}>
              <Text
                size="14px"
                className="product-text"
                uppercase
                bold
                color="#999"
              >
                CARTÃO DE CRÉDITO
              </Text>
              <Card>
                <Row>
                  <Col lg={12}>
                    <Input
                      dataCy="card"
                      defaultValue={cardNumber}
                      placeholder="____.____.____.____"
                      label="Número do cartão:"
                      mask="9999.9999.9999.9999"
                      onChange={e => setCardNumber(e.target.value)}
                      error={
                        !cardError &&
                        cardNumber &&
                        cardNumber.length !== 0 &&
                        typeof cardNumber !== 'undefined'
                          ? 'Numero de cartao invalido'
                          : null
                      }
                    />
                  </Col>
                  <Col lg={12}>
                    <Input
                      dataCy="name"
                      placeholder="Como no cartao"
                      defaultValue={cardName}
                      onChange={e => setCardName(e.target.value)}
                      label="Nome do titular:"
                      error={
                        typeof cardName !== 'undefined' && cardName.length === 0
                          ? 'Digite seu nome'
                          : null
                      }
                    />
                  </Col>
                  <Col lg={6} xs={6}>
                    <Input
                      dataCy="validity"
                      defaultValue={cardValidity}
                      placeholder="__/____"
                      mask="99/9999"
                      onChange={e => setCardValidity(e.target.value)}
                      label="Validade (mes/ano):"
                      error={
                        typeof cardValidity !== 'undefined' &&
                        cardValidity.length < 7
                          ? 'Data invalida'
                          : null
                      }
                    />
                  </Col>
                  <Col lg={6} xs={6}>
                    <Input
                      dataCy="ccv"
                      defaultValue={ccv}
                      onChange={e => setCcv(e.target.value)}
                      placeholder="____"
                      mask="999"
                      label="CVV:"
                      error={
                        typeof ccv !== 'undefined' && ccv.length < 3
                          ? 'CCV invalido'
                          : null
                      }
                    />
                  </Col>
                </Row>
              </Card>
            </Col>

            <Checkout data={data} />

            <Col lg={12} xs={12}>
              <Link
                to="/success"
                className="animated fadeIn"
                data-cy="buttonPayment"
              >
                <Button
                  disabled={handleFormValid() || !cardError}
                  onClick={handleSubmit()}
                >
                  Finalizar o pedido
                </Button>
              </Link>
            </Col>
          </Row>
        ) : (
          <Redirect to="/cart" />
        )}
      </Container>
    </ContainerStyled>
  );
};

export default DefaultLayout(connect()(Payment));
