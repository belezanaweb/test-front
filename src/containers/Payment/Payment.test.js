import React, { Suspense } from 'react'
import { BrowserRouter as Router } from 'react-router-dom'
import { render, screen, waitFor, fireEvent } from '@testing-library/react'
import { CartStore } from '../../store/CartStore'
import { PaymentStore } from '../../store/PaymentStore'
import Payment from './Payment'

const CartStoreValue = {
  state: {
    items: [
      {
        product: {
          name: 'Teste',
          priceSpecification: {
            price: 123.45
          },
          imageObjects: [
            {
              small: 'http://localhost.test'
            }
          ]
        }
      }
    ],
    subTotal: 123.45,
    total: 432.1,
    shippingTotal: 345.67,
    discount: 37.02
  }
}

const PaymentStoreValue = {
  state: {
    ccNumber: '****.****.****.1234',
    ccName: 'João da Silva',
    ccExpiry: '05/2019'
  },
  dispatch: {
    ccNumber: jest.fn(),
    ccName: jest.fn(),
    ccExpiry: jest.fn()
  }
}

test('renders Payment', async () => {
  const { container } = render(
    <Suspense fallback={<div>Testing...</div>}>
      <CartStore value={CartStoreValue}>
        <PaymentStore value={PaymentStoreValue}>
          <Payment />
        </PaymentStore>
      </CartStore>
    </Suspense>
  )

  await waitFor(() => screen.getByText('Finalizar o pedido'))

  expect(container).toMatchSnapshot()
})

test('Submits payment data', async () => {
  render(
    <Router>
      <Suspense fallback={<div>Testing...</div>}>
        <CartStore value={CartStoreValue}>
          <PaymentStore value={PaymentStoreValue}>
            <Payment />
          </PaymentStore>
        </CartStore>
      </Suspense>
    </Router>
  )

  await waitFor(() => screen.getByText('Finalizar o pedido'))

  fireEvent.change(screen.getByLabelText('Número do cartão:'), {
    target: { value: '1234123412341234' }
  })

  fireEvent.change(screen.getByLabelText('Nome do titular:'), {
    target: { value: 'João da Silva' }
  })

  fireEvent.change(screen.getByLabelText('Validade (mês/ano):'), { target: { value: '12/2022' } })

  fireEvent.change(screen.getByLabelText('CVV:'), { target: { value: '123' } })

  fireEvent.click(screen.getByText('Finalizar o pedido'))
})
