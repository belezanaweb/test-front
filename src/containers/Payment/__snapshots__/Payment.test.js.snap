// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Payment 1`] = `
.c22 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  padding: 18px;
  width: 100%;
  min-height: 60px;
  border: 0;
  border-radius: 3px;
  background-color: #ff6c00;
  color: #fff;
  font-size: 20px;
  font-weight: 700;
  -webkit-letter-spacing: 0.5px;
  -moz-letter-spacing: 0.5px;
  -ms-letter-spacing: 0.5px;
  letter-spacing: 0.5px;
  line-height: 1.2;
  text-align: center;
  -webkit-text-decoration: none;
  text-decoration: none;
  text-transform: uppercase;
  box-shadow: inset 0 -3px 0 0 #d45a00,0 2px 4px 0 rgba(0,0,0,0.25);
  cursor: pointer;
}

.c22:hover,
.c22:active {
  background-color: #d45a00;
  box-shadow: none;
}

.c7 {
  padding: 12px 12px 13px 13px;
  border-radius: 3px;
  background-color: #fff;
  box-shadow: 1px 1px 5px 0 rgba(0,0,29,0.22);
}

.c6 {
  display: block;
  margin: 0 0 5px 13px;
  color: #999;
  font-size: 14px;
  font-weight: 700;
  line-height: 1.21428571428571;
  text-transform: uppercase;
}

.c9 {
  display: grid;
  grid-gap: 5px;
}

.c14 {
  display: none;
  margin-top: 3px;
  color: #f30;
  font-size: 12px;
  line-height: 1.166666667;
}

.c12 {
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  height: 45px;
  padding: 13px;
  border: 1px solid #e7e7e7;
  border-radius: 3px;
  background-color: #fff;
  font-size: 16px;
  line-height: 1.1875;
  box-shadow: inset 0 1px 2px 0 rgba(0,0,0,0.2);
}

.c12::-webkit-input-placeholder {
  color: #e0e7ee;
}

.c12::-moz-placeholder {
  color: #e0e7ee;
}

.c12:-ms-input-placeholder {
  color: #e0e7ee;
}

.c12::placeholder {
  color: #e0e7ee;
}

.c12:focus {
  border-color: #a43287;
}

.c12:invalid:not(:placeholder-shown) {
  border-color: #f30;
}

.c12:invalid:not(:placeholder-shown) + .c13 {
  display: block;
}

.c11 {
  display: block;
  margin-left: 2px;
  color: #ccc;
  font-size: 12px;
  font-weight: 700;
  line-height: 1.16666666666667;
}

.c8 {
  display: grid;
  grid-template-columns: 52.45901% 1fr;
  grid-gap: 25px 20px;
}

.c10 {
  grid-column-end: span 2;
}

.c16 {
  display: grid;
  grid-gap: 9px;
  padding: 15px 13px 13px 14px;
  border: 1px solid #ccc;
  border-radius: 3px;
  color: #212122;
  font-size: 14px;
  line-height: 1.21428571428571;
}

.c17 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  -webkit-align-items: flex-start;
  -webkit-box-align: flex-start;
  -ms-flex-align: flex-start;
  align-items: flex-start;
  margin: 0;
}

.c18 {
  text-transform: uppercase;
}

.c19 {
  text-align: right;
  margin: 0;
}

.c20 {
  color: #ff7800;
}

.c21 {
  margin-top: 7px;
  font-weight: 700;
}

.c1 {
  width: 100%;
  margin-bottom: 10px;
  padding: 12px 0;
  background-color: #fff;
  box-shadow: 1px 1px 5px 0 rgba(0,0,29,0.22);
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  max-width: 1024px;
  margin: 0 auto;
  padding: 0 20px;
  list-style: none;
}

.c3 {
  display: block;
  color: #ccc;
  font-size: 13px;
  font-weight: 700;
  line-height: 1.23076923076923;
  text-transform: uppercase;
}

.c4 {
  display: block;
  color: #ccc;
  font-size: 13px;
  font-weight: 700;
  line-height: 1.23076923076923;
  text-transform: uppercase;
  color: #ff7800;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.c5 {
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
  display: grid;
  grid-gap: 20px;
  padding: 0 9px 20px;
  width: 100%;
  max-width: 1024px;
  margin: 0 auto;
}

.c15 {
  display: grid;
  -webkit-align-content: start;
  -ms-flex-line-pack: start;
  align-content: start;
  grid-gap: 20px;
}

@media only screen and (min-width:768px) {
  .c5 {
    margin-top: 40px;
    grid-template-columns: 1fr 340px;
  }
}

@media only screen and (min-width:768px) {
  .c15 {
    margin-top: 20px;
  }
}

<div>
  <main
    class="c0"
  >
    <div
      class="c1"
    >
      <ul
        class="c2"
      >
        <li
          class="c3"
        >
          Sacola
        </li>
        <li
          class="c4"
        >
          Pagamento
        </li>
        <li
          class="c3"
        >
          Confirmação
        </li>
      </ul>
    </div>
    <div
      class="c5"
    >
      <section
        class=""
      >
        <h1
          class="c6"
        >
          Cartão de Crédito
        </h1>
        <form
          class="c7 c8"
          id="_form"
          method="POST"
        >
          <div
            class="c9 c10"
          >
            <label
              class="c11"
              for="_input_credit_card"
            >
              Número do cartão:
            </label>
            <input
              autocomplete="cc-number"
              class="c12"
              id="_input_credit_card"
              name="cardnumber"
              pattern="\\\\d{4}\\\\.\\\\d{4}\\\\.\\\\d{4}\\\\.\\\\d{4}"
              placeholder="____.____.____.____"
              required=""
              value=""
            />
            <div
              class="c13 c14"
            >
              Número do cartão de crédito inválido
            </div>
          </div>
          <div
            class="c9 c10"
          >
            <label
              class="c11"
              for="_input_name"
            >
              Nome do titular:
            </label>
            <input
              autocomplete="cc-name"
              class="c12"
              id="_input_name"
              name="ccname"
              placeholder="Como no cartão"
              required=""
              value=""
            />
            <div
              class="c13 c14"
            >
              Invalid Field
            </div>
          </div>
          <div
            class="c9"
          >
            <label
              class="c11"
              for="_input_expiry_date"
            >
              Validade (mês/ano):
            </label>
            <input
              autocomplete="cc-exp"
              class="c12"
              id="_input_expiry_date"
              name="cc-exp"
              pattern="\\\\d{2}/\\\\d{4}"
              placeholder="__/____"
              required=""
              value=""
            />
            <div
              class="c13 c14"
            >
              Validade do cartão de crédito inválida
            </div>
          </div>
          <div
            class="c9"
          >
            <label
              class="c11"
              for="_input_cvc"
            >
              CVV:
            </label>
            <input
              autocomplete="cc-csc"
              class="c12"
              id="_input_cvc"
              name="cvc"
              pattern="\\\\d{3}"
              placeholder="___"
              required=""
              value=""
            />
            <div
              class="c13 c14"
            >
              CVV do cartão de crédito inválido
            </div>
          </div>
        </form>
      </section>
      <aside
        class="c15"
      >
        <section
          class="c16"
        >
          <dl
            class="c17"
          >
            <dt
              class="c18"
            >
              Produtos
            </dt>
            <dd
              class="c19"
            >
              <span>
                R$ 123.45
              </span>
            </dd>
          </dl>
          <dl
            class="c17"
          >
            <dt
              class="c18"
            >
              Frete
            </dt>
            <dd
              class="c19"
            >
              <span>
                R$ 345.67
              </span>
            </dd>
          </dl>
          <dl
            class="c17 c20"
          >
            <dt
              class="c18"
            >
              Desconto
            </dt>
            <dd
              class="c19"
            >
              − 
              <span>
                R$ 37.02
              </span>
            </dd>
          </dl>
          <dl
            class="c17 c21"
          >
            <dt
              class="c18"
            >
              Total
            </dt>
            <dd
              class="c19"
            >
              <span>
                R$ 432.10
              </span>
            </dd>
          </dl>
        </section>
        <button
          class="c22"
          form="_form"
          type="submit"
        >
          Finalizar o pedido
        </button>
      </aside>
    </div>
  </main>
</div>
`;
